.github/workflows/iroha2-dev-nightly.yml

```diff
@@ -3,17 +3,26 @@ name: I2::Dev::Nightly::Publish
 on: workflow_dispatch
 
 env:
+  RUST_TOOLCHAIN: nightly-2024-04-18
   DOCKER_COMPOSE_PATH: defaults
 
 jobs:
   build_executor:
-    runs-on: ubuntu-latest
-    container:
-      image: hyperledger/iroha2-ci:nightly-2024-04-18
+    runs-on: macos-latest
     timeout-minutes: 30
     steps:
       - uses: actions/checkout@v4
-      - uses: Swatinem/rust-cache@v2
+      - uses: dtolnay/rust-toolchain@master
+        with:
+          toolchain: ${{ env.RUST_TOOLCHAIN }}
+          targets: aarch64-apple-darwin,wasm32-unknown-unknown
+          # needed to build smart contracts
+          components: rust-src
+      - uses: Leafwing-Studios/cargo-cache@v2
```

Comment:
how's it different/better than `rust-cache`?

Reply:
`rust-cache` never hits the cache for some reason